qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -256,12 +301,35 @@ def compute_minimum_eigenvalue(
             )
             self._excitation_list.append(self._excitation_pool[max_grad_index])
             theta.append(0.0)
-            # run VQE on current Ansatz
+            # setting up the ansatz for the VQE iteration
             self._tmp_ansatz.operators = self._excitation_list
             self.solver.ansatz = self._tmp_ansatz
             self.solver.initial_point = theta
+            # evaluating the eigenvalue with the internal VQE
+            prev_raw_vqe_result = raw_vqe_result
             raw_vqe_result = self.solver.compute_minimum_eigenvalue(operator)
             theta = raw_vqe_result.optimal_point.tolist()
+            # checking convergence based on the change in eigenvalue
+            if iteration > 1:
+                eigenvalue_diff = np.abs(raw_vqe_result.eigenvalue - history[-1])
+                if eigenvalue_diff < self.eigenvalue_threshold:
+                    logger.info(
+                        "AdaptVQE terminated successfully with a final change in eigenvalue: %s",
+                        str(eigenvalue_diff),
+                    )
+                    termination_criterion = TerminationCriterion.CONVERGED
+                    logger.debug(
```

Comment:
Since we already did all the calculations to get the update step, wouldn't generally give a better energy to keep it anyways?

Reply:
I see what you mean. The idea here was that if we want to stick to as small of an ansatz as possible to achieve a desired configuration this new result would reflect an ansatz that is just slightly too large. Thus I reverted back to one prior step. Does that make sense?